training/src/main/scala/com/airbnb/aerosolve/training/TrainingUtils.scala

```diff
@@ -256,17 +265,32 @@ object TrainingUtils {
       .groupBy($"family", $"feature")
       .agg(
         count(lit(1)) as "count",
-        min($"value"),
-        max($"value"),
-        sum($"value"),
-        // TODO: use built-in stdev function in Spark 1.6
-        sum($"value" * $"value")
+        Seq(
+          min($"value"),
+          max($"value"),
+          sum($"value"),
+          // TODO: use built-in stdev function in Spark 1.6
+          sum($"value" * $"value")
+        ) ++
+          // TODO: use built-in percentileApprox in Spark 2.0
+          quantiles.map(q => callUDF("percentile_approx", $"value", lit(q))): _*
```

Comment:
It's a type casting. The function takes `columns: Column*` for argument. What we have is `Seq[Column]`. Normally you would do `columns: Column*` but in this case, you can just do `columns: _*` since `Column` type can be easily inferred. http://stackoverflow.com/questions/6051302/what-does-colon-underscore-star-do-in-scala